### PR TITLE
[16.0] odoo_repository: option to force cloned repository name

### PR DESCRIPTION
**Pull request by sebalix**
New option `clone_name` to force the cloned repository folder name. We cannot use existing `name` field as it is used as the real repo name to call GH API for instance.

Short story: this is to workaround an issue encountered on our mounted filesystem, the name of the repository is not usable anymore as cloned repository folder.

### CODE REVIEW

**Comment by hparfr** [comment]
I do not know your requirements and I didn't pay attention to this part of the code
But if the path is  `/org_name/repo_name` we may have more unicity  ?

**Comment by sebalix** [comment]
By default the cloned repository path is `{somewhere}/{org}/{repo_name}` as you mention, the same parameters are used by the migration scanner (through `oca-port`) to request GH API.

Here we have an issue on our network mounted filesystem where some `{somewhere}/{org}/{repo_name}` paths are not usable anymore because of ghost files we cannot removed, and we cannot clone in such path anymore. Changing the `<odoo.repository>.name` to clone it somewhere else will work, but the GH API call cannot work anymore as this repo name won't exist on GH.

Here this allows to differentiate the repository name on GH, and the folder name where we clone it on disk.
At first I wanted to make all of this specific in our project, but it wasn't easy to override all scanners to add this `clone_name` parameter. At the end better to give this control to the scanner anyway.

EDIT: about unicity, with this new field we could indeed break it, like having a repo `OCA/my_repo` that will be cloned/fetched in `OCA/server-tools` path (with `clone_name = "server-tools"`) :thinking:

**Comment by hparfr** [comment]
Ok, fair enough :)
May be add a bit of explaination of why this feature is here and when it should be removed.

File: odoo_repository/models/odoo_repository.py [hunk 1]
```diff
@@ -53,6 +53,12 @@ class OdooRepository(models.Model):
         string="Clone URL",
         help="Used to clone the repository.",
     )
+    clone_name = fields.Char(
+        help=(
+            "Force the name of the cloned repository (folder on disk). "
```
Comment (hparfr) [inline odoo_repository/models/odoo_repository.py:58]:
```suggestion
            "Force the name of the cloned repository (folder on disk)."
```

**Review by hparfr — APPROVED** [review]

**Comment by sebalix** [comment]
But yes, this `clone_name` option is legit at the scanner level I think (user of the scanner should have the control where to scan the repo), but in `odoo.repository` this option should be hidden ideally (maybe add `base.group_no_one` on this field).

File: odoo_repository/models/odoo_repository.py (hunk 1 above)
Comment (sebalix) [reply odoo_repository/models/odoo_repository.py:58]: This space is here on purpose, there is a sentence right after that gets concatenated :) ("If not sent...")

Comment (hparfr) [reply odoo_repository/models/odoo_repository.py:58]: ah ah ah :) Indeed sorry for the noise

**Pull request merged**